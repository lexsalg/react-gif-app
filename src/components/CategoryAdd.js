import { useState } from "react";

import PropTypes from 'prop-types';

export const CategoryAdd = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            // setCategories([...categories, 'nuevo'])
            setCategories(state => [inputValue, ...state])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>

            <h1>{inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}


CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}