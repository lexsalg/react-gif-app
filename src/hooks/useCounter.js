import { useState } from 'react'


export const useCounter = (initialState = 10) => {
    const [state, setState] = useState(initialState);

    const increment = () => setState(state + 1);
    const decrement = () => setState(state - 1);
    const factorAdd = (factor) => setState(state + factor);
    const factorSub = (factor) => setState(state - factor);
    const reset = () => setState(initialState);



    return { state, increment, decrement, factorAdd, factorSub, reset };

}
