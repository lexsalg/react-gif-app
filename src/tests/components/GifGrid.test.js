import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el <GifGrid />', () => {

    const category = 'One Punch';

    test('should be displayed correctly', () => {

        useFetchGifs.mockReturnValue({ data: [], loading: true });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    })


    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const data = [{ id: '11', url: 'https://localhost/123.jpg', title: 'titulo' }];

        useFetchGifs.mockReturnValue({ data, loading: false });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(data.length);


    })



})
