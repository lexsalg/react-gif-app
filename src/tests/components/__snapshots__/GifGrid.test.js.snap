// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="11"
      key="11"
      title="titulo"
      url="https://localhost/123.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> should be displayed correctly 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
